### DISCUSSION TITLE
No WireGuard "Save settings" "Stop WireGuard" buttons

### DISCUSSION THREAD

**Discussion by HansiJung** [RaspAP/raspap-webgui#1430] [Q&A]
Hi,
I just set up a new Raspberry Pi 3 B with RaspAP V2.9.6.

In the WireGuard Dash I can upload my configuration but in the bottem line I can not see any "Save settings" "Stop WireGuard" buttons.

In the upper left corner wg up is not green.

Aftre reboot I can see on the other Server that the WireGurad is conected, but on my RaspAP it is not up.

Any idea?

**Comment by billz** [comment]
If the save / stop buttons aren't appearing, it's possible that something related to your config is creating a PHP error. Try executing the following in the shell:

```
sudo tac /var/log/lighttpd/error.log | grep -m 1 'PHP Fatal error'
```

**Comment by HansiJung** [comment]
Ahh, there ia something

`2023-10-23 17:36:10: (mod_fastcgi.c.449) FastCGI-stderr:PHP Fatal error:  Uncaught TypeError: htmlspecialchars(): Argument #1 ($string) must be of type string, array given in /var/www/html/templates/wg/general.php:96
`

**Reply by billz** [reply]
This part of the template outputs the DNS value `$wg_srvdns` that is parsed from your wg config. DNS values usually take a form like this:

```
[Interface]
Address = 10.0.44.2/32
PrivateKey = [private_key]
DNS = 1.1.1.1
```

RaspAP is interpreting this setting as an array, instead of a single string value. It should catch this before passing it to the template, but I'd start by checking the value in your wg0.conf file.

**Comment by HansiJung** [comment]
Yersterday there was an update to V2.9.7.

I reinstalled it.

Still the same problem

**Reply by billz** [reply]
> start by checking the value in your wg0.conf file.

**Reply by HansiJung** [reply]
It looks like this:

```
[Interface]
PrivateKey = [PrivateKey]
Address = 192.168.178.209/32
DNS = 192.168.178.1

[Peer]
PublicKey = [PublicKey]
PresharedKey = [PresharedKey]
AllowedIPs = 0.0.0.0/0
Endpoint = [Endpoint]
PersistentKeepalive = 25
```

**Reply by billz** [reply]
Yep, it's a bug. Thanks for reporting this
https://github.com/RaspAP/raspap-webgui/pull/1432
